python/tests/test_writer.py

```diff
@@ -310,3 +322,51 @@ def test_writer_fails_on_protocol(existing_table: DeltaTable, sample_data: pa.Ta
     existing_table.protocol = Mock(return_value=ProtocolVersions(1, 2))
     with pytest.raises(DeltaTableProtocolError):
         write_deltalake(existing_table, sample_data, mode="overwrite")
+
+
+def test_writer_with_max_rows(tmp_path: pathlib.Path):
+    def get_multifile_stats(table: DeltaTable) -> Iterable[Dict]:
+        log_path = get_log_path(table)
+
+        # Should only have single add entry
+        for line in open(log_path, "r").readlines():
+            log_entry = json.loads(line)
+
+            if "add" in log_entry:
+                yield json.loads(log_entry["add"]["stats"])
+
+    data = pa.table(
+        {
+            "colA": pa.array(range(0, 1000), pa.int32()),
+            "colB": pa.array([i * random.random() for i in range(0, 1000)], pa.float64())
+        }
+    )
+    path = str(tmp_path)
+    write_deltalake(path, data, file_options=ParquetFileFormat().make_write_options(), max_rows_per_file=100, max_rows_per_group=100)
+
+    table = DeltaTable(path)
+    stats = get_multifile_stats(table)
+
+    assert sum([stat_entry['numRecords'] for stat_entry in stats]) == 1000
+
+
+def test_writer_with_options(tmp_path: pathlib.Path):
+    expected_column_values = [int(datetime(year_, 1, 1, 0, 0, 0).timestamp() * 1e6) for year_ in range(9000, 9010)]
+    data = pa.table(
+        {
+            "colA": pa.array(expected_column_values, pa.timestamp('us'))
+        }
+    )
+    path = str(tmp_path)
+    opts = ParquetFileFormat().make_write_options().update(
+        compression='GZIP',
+        coerce_timestamps='us'
+    )
+    write_deltalake(path, data, file_options=opts)
+
+    table = DeltaTable(path).to_pyarrow_dataset(parquet_read_options=ParquetReadOptions(coerce_int96_timestamp_unit='us')).to_table().to_pydict()['colA']
+
+    # since either session (Spark) or system timezone is used when reading timestamps back into python
+    # a conversion is required for this synthetic case. Should not be an issue with a real table, as long as a system/Spark session timestamp matches the one used when data was written
+
+    assert [int(d.timestamp() * 1e6 - time.timezone * 1e6) for d in table] == expected_column_values
```

Comment:
Above, if you just let table be a `pa.Table`, you could simply make this assertion `assert table == expected` (or `data`).

Reply:
applied, mourn all the time i spent figuring out timezone issue :)